kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/store/DbDataStore.java

```diff
@@ -342,12 +342,16 @@ private synchronized DataMessage storeInternal(String topic, byte[] payload, int
                 pstmt.setInt(6, -1);                                                // publishedMessageId
                 pstmt.setTimestamp(7, null);                                        // confirmedOn
 
-                if (payload.length < PAYLOAD_BYTE_SIZE_THRESHOLD) {
-                    pstmt.setBytes(8, payload);                                     // smallPayload
+                // smallPayload (=8) vs. largePayload (=9)
+                if (payload == null) {
```

Comment:
Can't we have for the null payload the cose as it was in ESF 7? https://github.com/eclipse/kura/blob/d78e70cff2404e292cb9876344d171273966a66c/kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/store/DbDataStore.java#L320

Reply:
so the is would be 
```
if (isNull(payload) || payload.length < PAYLOAD_BYTE_SIZE_THRESHOLD){
```

Reply:
Like this?

```java
if (payload == null || payload.length < PAYLOAD_BYTE_SIZE_THRESHOLD) {
    pstmt.setBytes(8, payload);
    pstmt.setNull(9, Types.BLOB);
} else ...
```

Reply:
Yes. I have put the isNull to have it clean. But works too in the way you have put it

Reply:
Yes.